Wizardry/Core/BattleSys/Source/BattleHit.c

```diff
@@ -212,15 +211,35 @@ void BattleGenerateHitAttributes(struct BattleUnit * attacker, struct BattleUnit
         else
         {
 #if (defined(SID_Pavise) && (SID_Pavise < MAX_SKILL_NUM))
-            if (CheckBattleSkillActivte(defender, attacker, SID_Pavise, GetUnitSkill(_tmpunit)))
+            if (CheckBattleSkillActivte(defender, attacker, SID_Pavise, GetUnitSkill(unit_def)))
             {
                 RegisterTargetEfxSkill(GetBattleHitRound(gBattleHitIterator), SID_Pavise);
                 gBattleStats.damage = 0;
             }
 #endif
         }
     }
+    if (gBattleStats.damage > 0)
+    {
+#if (defined(SID_KeenFighter) && (SID_KeenFighter < MAX_SKILL_NUM))
+        if (SkillTester(unit_def, SID_KeenFighter))
+        {
+            if (CheckCanTwiceAttackOrder(defender,attacker))
+            {
+                RegisterTargetEfxSkill(GetBattleHitRound(gBattleHitIterator), SID_KeenFighter);
```

Comment:
1. Better not to let them show the anim since they are not triggered by chance. These skills is also not given skill anim info.
2. KeenFighter and DragonSkin seems better not to be effective simultaneously, it is recommanded to use if..else to let only one skill to be active at a time

Reply:
1. I think feedback is important enought to overlook the lack of proc, otherwise there is zero communication to the player for why the attack is cut in half (this is how it is done in official game, wich is why I did it that way). As for Keen Fighter, it fall into the Vantage/Wary Fighter category where it is a conditional proc instead of a random one. The anim info has been added
2. Done